### DISCUSSION TITLE
[FEATURE] Cost of Higher Level Spells

### DISCUSSION THREAD

**Discussion by Kharmans** [misthero/dnd5e-spellpoints#127] [Q&A]
I don't know if you can make this on the module, since it's something I use only on my tables and it's not a rule on D&D5e, but since there's a part of homebrew rules already on the module why not give a try.

It's possible to configure an option that, when you cast spells of 6º level or above, the cost of spell points would double? Like, for the first time I cast a 6º level spell, it would have the cost of 9 spell points, but the second time I cast a 6º level spell, it would cost 18 spell points. For the third cast, 36 spell points, and so on. The same with spells of 7º, 8º and 9º level.

Can be done?

**Comment by misthero** [comment]
that's a good question with no easy solution, I've been thinking. 

You can enter formulas in the spell cost field, but something should count how many spells you've  already launched. The only solution I can think of, with the current tools and no other modules could be this, for level 6 slots the cost should be something like:

`9*(1+ @spells.spell6.max - @spells.spell6.value)`

this means that if you are full of level 6 spell slots the formula woud be: 9*1 (since your max slots and current slots are equal)

if you consume a 6th level spell your `@spells.spell6.value` would be 1 less than the max, and the formula would return 18 (9*2) and so on.

BUT...

for this to work your players should check the "consume slot" checkbox every time they cast a spell. Also if you want to allow like 20 6th level spells per rest you should increase the number of slots.

**Comment by Kharmans** [comment]
Hummmmm... that's a good solution! I can configure a max of spell slots of 3 for 6º spell, since thats the maximum of casts a player can do with 133 spell points.

**1º cast**: 9 points for a total of 124 spellpoints;
**2º cast**: 18 points for a total of 106 spellpoints;
**3º cast**: 36 points for a total of 70 spellpoints.

Another cast can't be done cause it would cost 72 spellpoints and the player would'n have that. And I would do the same for spells of 7º, 8º and 9º levels, cause their also permit only 3 casts.

The thing is... how can I adjust the maximum of spell slots? I don't remember seeing this on Foundry.

**Comment by misthero** [comment]
> The thing is... how can I adjust the maximum of spell slots? I don't remember seeing this on Foundry.

![immagine](https://github.com/misthero/dnd5e-spellpoints/assets/3662610/132abcfc-dc17-4f8b-8717-97f30c04a74a)

**Comment by Kharmans** [comment]
Oh, I know how to add manually, I am talking about add them by default.

**Comment by misthero** [comment]
> Oh, I know how to add manually, I am talking about add them by default.

I don't know if a module exists for that, you could probably create a macro to update the actors with one click, but that's out of the scope of this module. May I suggest looking for help in the macro channel on the foundry discord server?

https://discord.com/channels/170995199584108546/699750150674972743
